### ISSUE TITLE
Tye init --force omit defaults behavior when tye.yaml already exist

### ISSUE THREAD

**Issue by unaizorrilla** [dotnet/tye#207]
### Describe the bug
**Tye init --force** does not omit default values on serialization when tye.yaml already exist. 

When tye init is executed at the first time the yaml  result is:

```yaml
# tye application configuration file
# read all about it at https://github.com/dotnet/tye
#
# when you've given us a try, we'd love to know what you think:
#    https://aka.ms/AA7q20u
#
name: tye-samples
services:
- name: frontend
  project: FrontEnd/FrontEnd.csproj
- name: api
  project: Api/Api.csproj
```
If tye init --force is executed again, some default values are not omited  and result file include default values:

```yaml
# tye application configuration file
# read all about it at https://github.com/dotnet/tye
#
# when you've given us a try, we'd love to know what you think:
#    https://aka.ms/AA7q20u
#
name: tye-samples
services:
- name: frontend
  project: FrontEnd/FrontEnd.csproj
  bindings:
  - autoAssignPort: true
    protocol: https
  - autoAssignPort: true
    protocol: http
  volumes: []
  env: []
- name: api
  project: Api/Api.csproj
  bindings:
  - autoAssignPort: true
    protocol: https
  - autoAssignPort: true
    protocol: http
  volumes: []
  env: []
ingress: []
```

The serializer is configured correctly in both cases:
 
**ConfigureDefaultValuesHandling(DefaultValuesHandling.OmitDefaults)**

But the application object from ConfigFactory.FromFile

https://github.com/dotnet/tye/blob/2a1b22d4bd911ef4ce28845ac49cbd620727be1d/src/tye/InitHost.cs#L60 

have null values for defaults when tye.yaml does not exist and not null,empty arrays, when tye.yaml exist. For yamldotnet serializer empty array is not a default value and is not omited.

### To Reproduce
tye init
tye init --force

**Comment by unaizorrilla** [comment]
Hi @davidfowl 

The label "good first issue" is like "up for grabs" ? can i do a pr for this?

**Comment by davidfowl** [comment]
Yep, please send a PR 😄

**Comment by rynowak** [comment]
Yes, please do :)

**Comment by unaizorrilla** [comment]
Fixed with #237
